File: unix/boot/spp/rpp/ratlibf/patsiz.f
      integer function patsiz (pat, n)
      integer pat (128)
      integer n
      if (.not.(pat (n) .eq. 97 .or. pat (n) .eq. 123 .or. pat (n) .eq. 
     *125))goto 23000
      patsiz = 2
      goto 23001
23000 continue
      if (.not.(pat (n) .eq. 37 .or. pat (n) .eq. 36 .or. pat (n) .eq. 6
     *3))goto 23002
      patsiz = 1
      goto 23003
23002 continue
      if (.not.(pat (n) .eq. 91 .or. pat (n) .eq. 110))goto 23004
      patsiz = pat (n + 1) + 2
      goto 23005
23004 continue
      if (.not.(pat (n) .eq. 42))goto 23006
      patsiz = 4
      goto 23007
23006 continue
      call error (24Hin patsiz: can't happen.)
23007 continue
23005 continue
23003 continue
23001 continue
      return
      end

File: iraf/unix/boot/spp/rpp/ratlibr/patsiz.r
include	defs

# patsiz --- returns size of pattern entry at pat (n)

   integer function patsiz (pat, n)
   character pat (MAXPAT)
   integer n

   if (pat (n) == CHAR | pat (n) == START_TAG | pat (n) == STOP_TAG)
      patsiz = 2
   else if (pat (n) == BOL | pat (n) == EOL | pat (n) == ANY)
      patsiz = 1
   else if (pat (n) == CCL | pat (n) == NCCL)
      patsiz = pat (n + 1) + 2
   else if (pat (n) == CLOSURE)      # optional
      patsiz = CLOSIZE
   else
      call error ("in patsiz: can't happen.")

   return
   end
